core/src/scene/light.h

```diff
@@ -4,20 +4,21 @@
 #include "glm/glm.hpp"
 #include "util/shaderProgram.h"
 
-typedef enum {
-    DEFAULT,
-    VERTEX,
-    FRAGMENT
-} InjectionType;
-
 enum class LightType {
     UNDEFINED,
+    AMBIENT,
     DIRECTIONAL,
     POINT,
     SPOT,
     CUSTOM
 };
 
+enum class LightOrigin {
+    CAMERA,
+    GROUND,
+    WORLD
```

Comment:
I see, thanks. For the first pass we can take the easy route and drop straight down from the camera. We should experiment with projecting from the center from the screen onto the ground too, it's not obvious to me a priori that one is always better.